### PR TITLE
Fix bootloader tests

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

File: subsys/bootloader/Kconfig
Comment (hakonfam) [inline subsys/bootloader/Kconfig:122]: We typically don't do `default n` since its the default :smile:

File: tests/subsys/bootloader/bl_validation_ff_key/testcase.yaml
Comment (hakonfam) [inline tests/subsys/bootloader/bl_validation_ff_key/testcase.yaml:3]: We can remove support for the PDK while we are here

File: tests/drivers/fprotect/negative/testcase.yaml
Comment (hakonfam) [inline tests/drivers/fprotect/negative/testcase.yaml:3]: as befare, we should remove pdk here since support for it will be removed

File: tests/drivers/fprotect/negative/dummy_huk.hex
Comment (kyberdin) [inline tests/drivers/fprotect/negative/dummy_huk.hex:1]: I see this file used in the associated `CMakeLists.txt`, but do we normally commit hex files for tests?

File: subsys/bootloader/Kconfig
Comment (kyberdin) [inline subsys/bootloader/Kconfig:120]: This is for debug testing that users of NCS will never need to perform, correct? Should we remove these from menuconfig visibility then without some backend Kconfig option that enables Nordic debug testing?

File: tests/subsys/bootloader/bl_validation_ff_key/provision_ff.hex
Comment (kyberdin) [inline tests/subsys/bootloader/bl_validation_ff_key/provision_ff.hex:1]: Same thing here about committing hex files, but where is this used?

**Review by ChristopherBittner — APPROVED** [review]

**Review by Vge0rge — APPROVED** [review]

File: subsys/bootloader/Kconfig
Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig:120]: That would be good, but I don't think we can. If we remove the prompt, they will not be modifyable via prj.conf.

File: tests/subsys/bootloader/bl_validation_ff_key/provision_ff.hex
Comment (oyvindronningstad) [reply tests/subsys/bootloader/bl_validation_ff_key/provision_ff.hex]: Thanks, it was left over from an earlier iteration. Removed

File: tests/drivers/fprotect/negative/dummy_huk.hex
Comment (oyvindronningstad) [reply tests/drivers/fprotect/negative/dummy_huk.hex]: This commit is removed since it requires another PR. The reason for commiting the hex file is for convenience. It's the easiest way to place the data at that location in flash. I think this is unproblematic since it builds out of the box, and it's only a test.

File: tests/drivers/fprotect/negative/testcase.yaml
Comment (oyvindronningstad) [reply tests/drivers/fprotect/negative/testcase.yaml]: Done

File: subsys/bootloader/Kconfig
Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: Done

**Comment by oyvindronningstad** [comment]
Removed the HUK loading test.

**Review by hakonfam — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
@ioannisg @tejlmand I guess I need a code owner approval.

**Review by kyberdin — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

File: subsys/bootloader/Kconfig
Comment (tejlmand) [reply subsys/bootloader/Kconfig:120]: You can make a `tests/subsys/bootloader/bl_validation_ff_key/Kconfig` fil, which contains:
```
config SB_DEBUG_NO_VERIFY_HASHES
	bool
	default y

source "Kconfig.zephyr"
```

so please make this symbol promptless.

File: subsys/bootloader/Kconfig
Comment (tejlmand) [inline subsys/bootloader/Kconfig:121]:
```suggestion
	bool
```

File: tests/subsys/bootloader/bl_validation_ff_key/prj.conf
Comment (tejlmand) [inline tests/subsys/bootloader/bl_validation_ff_key/prj.conf:10]: Please use a Kconfig file for this, and remove the prompt.
```suggestion
```

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: subsys/bootloader/Kconfig
Comment (ioannisg) [reply subsys/bootloader/Kconfig:120]: @oyvindronningstad ^^

**Review by ioannisg — APPROVED** [review]
Just a question @oyvindronningstad , why the _ns target is removed from platform_allow?

**Comment by oyvindronningstad** [comment]
> why the _ns target is removed from platform_allow?

@ioannisg because b0+spm+app doesn't really work. I could make a variant with b0+mcuboot+spm+app, but since this is just testing that b0 fails I don't see a big reason to test it with that many configurations.

File: subsys/bootloader/Kconfig
Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig:120]: Thanks @tejlmand, fixed.

**Comment by oyvindronningstad** [comment]
@tejlmand Can you take another look?

**Review by tejlmand — APPROVED** [review]

**Pull request merged**
